### ISSUE TITLE
libftdi API update

### ISSUE THREAD

**Issue by yoe** [OpenLightingProject/ola#1012]
Hi,

The Debian maintainer of libftdi filed a [bug](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=810374) against ola. I tried the simple fix ("s/libftdi-dev/libftdi1-dev/" over debian/control), but that results in no FTDI plugin being compiled.

Someone will need to look at the changes that were made in the new FTDI library and update ola accordingly. Mean time, I'll have to still compile against the old library.

**Comment by peternewman** [comment]
@Keeper-of-the-Keys mentioned this on IRC today. Trusty (Ubuntu 14.04) is in LTS until April 2019 and only has libfdti-dev:
https://packages.ubuntu.com/search?keywords=libftdi&searchon=names&suite=trusty&section=all

So we should probably try and support both versions until then at least.

**Comment by peternewman** [comment]
I've made an initial start here:
https://github.com/OpenLightingProject/ola/compare/0.10...peternewman:0.10-libftdi1

It doesn't compile yet:
https://travis-ci.org/peternewman/ola/jobs/349129800

**Comment by yoe** [comment]
That won't solve the issue, it will only postpone it. The only way to really fix it is to update the use of the library to the new version.

**Comment by Keeper-of-the-Keys** [comment]
I hope to submit the pull request by the end of the month.

2018-04-25 9:52 GMT+03:00 Wouter Verhelst <notifications@github.com>:

**Comment by peternewman** [comment]
@desert and @Keeper-of-the-Keys as mentioned above, I'd started hacking in some basic support, but it doesn't currently compile (so you may want to base it off that @Keeper-of-the-Keys ):
https://github.com/OpenLightingProject/ola/compare/0.10...peternewman:0.10-libftdi1

@desert I'm confused by the libusb-config/pkg-config, given we already use pkg-config for the FTDI stuff.

**Comment by desert** [comment]
Seeing the error seems the API just changed.

**Comment by peternewman** [comment]
Yep, it just needs a pile of ifdefs, like started here:
https://github.com/OpenLightingProject/ola/compare/0.10...peternewman:0.10-libftdi1#diff-de8c68bf126abbf38585da366b582faf

I did the initial stuff remotely just with Travis, but this seemed a point where stopping and doing it properly would make more sense and I've not had time yet.

**Comment by desert** [comment]
I have succesfully built ola with @Keeper-of-the-Keys libftdi1 branch. I used:

`./configure --disable-all-plugins --disable-e133 --disable-python-libs --disable-examples --disable-http --enable-ftdidmx `

I have not really tested any ftdi device. Will do that and inform.

Relevant run output:
```
olad/PluginManager.cpp:200: Started E1.31 (sACN)
olad/PluginManager.cpp:195: Trying to start USB
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 2:5 @0x55eac68731f0 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 2:4 @0x55eac686d070 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 2:3 @0x55eac686b620 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 2:2 @0x55eac6873150 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 2:1 @0x55eac6871cb0 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 6:1 @0x55eac6869110 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 5:11 @0x55eac6867510 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 5:10 @0x55eac686f890 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 5:9 @0x55eac6865960 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 5:2 @0x55eac6871ef0 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 5:1 @0x55eac686f9f0 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 1:2 @0x55eac686e760 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 1:1 @0x55eac686a7f0 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 4:1 @0x55eac686cee0 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 3:2 @0x55eac6867d50 [add]
libs/usb/HotplugAgent.cpp:166: USB hotplug event: 3:1 @0x55eac6869320 [add]
libs/usb/LibUsbThread.cpp:50: -- Starting libusb thread
common/thread/Thread.cpp:194: Thread , policy SCHED_OTHER, priority 0
libs/usb/LibUsbThread.cpp:35: ----libusb event thread is running
olad/PluginManager.cpp:200: Started USB
olad/PluginManager.cpp:195: Trying to start FTDI USB DMX
plugins/ftdidmx/FtdiWidget.cpp:124: Found 0 FTDI devices with PID: 24577.
plugins/ftdidmx/FtdiWidget.cpp:124: Found 0 FTDI devices with PID: 24593.
olad/PluginManager.cpp:200: Started FTDI USB DMX
```

**Comment by Keeper-of-the-Keys** [comment]
Peter - I originally did this against the 0.10.5 tag but then rebased to master so unless my understanding of git is wrong it should be possible to roll back to the commit before the rebase if that is better.

If you'd rather I'll see about pulling your branch first and then building with that.

I have also succesfully built this on my Ubuntu 18.04 laptop, it properly detects my FTDI boards I didn't have DMX equipment to connect to it but hope to test with a few LEDs tonight.

I only have 2 4 channel FTDI boards I would like to add a 2 channel and 1 channel so that I may test them together and separately.

**Comment by peternewman** [comment]
Thanks @Keeper-of-the-Keys .

I think I'd probably like the Travis testing stuff in the PR (ideally as well as credit for the bits I'd done, although it's fairly minor in this case). So yes, ideally if you can pull my branch and add the additional bits to finish the job, or I can pull the relevant changes you'd done into my branch and you/@desert could test against that? Whichever is easier.

Then we can try and get 0.10.7 out fairly soon with this fix in, as it seems fairly minor.

**Comment by desert** [comment]
Bring it on, I can test native x86 and yocto builds.

**Comment by desert** [comment]
I had some trouble setting up yocto cross compilations for foreign branches, a branch into the ola project would be easier.

**Comment by Keeper-of-the-Keys** [comment]
@peternewman I pushed a small change an hour ago that I hope will fix the travis issue, but travis isn't running as far as I can tell due to: "GitHub payload is missing a merge commit (mergeable_state: "unknown", merged: false) "

**Comment by peternewman** [comment]
@Keeper-of-the-Keys I suspect the Travis error was down to the fact the GitHub PR was in conflict. I used the web UI to resolve conflicts, but you can also do it on the command line. It's now attempting builds again.

**Comment by peternewman** [comment]
Closed by #1422 . See also for historic reference #394 .

@desert feel free to test please (the 0.10 branch)! @yoe this will be in 0.10.7 , probably releasing soon.

**Comment by desert** [comment]
Awesome, tomorrow will start testing.
